### DISCUSSION TITLE
HPC performance is better with separation of temporal nipype workdir (/tmp) and final HALFpipe workdir

### DISCUSSION THREAD

**Discussion by FeHoff** [HALFpipe/HALFpipe#406] [General] [answered]
For highly parallel workflows on compute clusters the HALFpipe workdir can blow up, as fMRIprep writes thousands of temporal files. If a central storage is used (as intended by HALFpipe) network I/O will limit performance. Is there a build-in solution for this?

**Comment by HippocampusGirl** [comment, marked as answer]
Hi @FeHoff, My personal solution is to llimit the number of jobs that run simultaneously and also to enable the automated cleanup of intermediate files with the `--keep none` command line flag. That usually keeps disk usage quite low.

**Reply by FeHoff** [reply]
I will use your solution for now, as the sample is not to big with n=200

**Comment by FeHoff** [comment]
Thanks for the quick answer!! I was thinking of a workaround, by using tmp/ on the compute nodes and combing finished HALFpipe workdirs in the central place, but the hardcoded paths in the compute graphs would prevent that, right?

**Reply by HippocampusGirl** [reply]
Exactly, the absolute paths need to stay the same for [Nipype](https://github.com/nipy/nipype) to be happy, at least in the way we currently implement things
